components/brave_ads/browser/ads_service_factory.cc

```diff
@@ -31,6 +32,10 @@ namespace brave_ads {
 // static
 AdsService* AdsServiceFactory::GetForProfile(
     Profile* profile) {
+  if (profile->IsOffTheRecord() || brave::IsTorProfile(profile)) {
+    return nullptr;
+  }
```

Comment:
@tmancey How are the changes in this file related to this PR? This was cleaned up before at https://github.com/brave/brave-core/pull/3597/files#diff-3639f92629488e9f547d8ff7ead4a5f7, and I'm not sure why we need to change it here for this PR's purpose.

Reply:
The changes are related as for Tor Profiles the ads service was started and ad notifications were shown. Ad notifications should not be shown for private tabs.

Reply:
if `profile->IsOffTheRecord()` doesn't handle Tor profiles, then it looks like we are in trouble, given how many other services could be started for Tor profiles? @yrliou